packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import './Signature.scss';
+
+const LineWidths = {
+  EXTRAFINE: 1,
+  FINE: 2,
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+
+
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: undefined,
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor() {
+    super();
+
+    this.state = { lineSegments: [] };
+  }
+
+  componentDidMount() {
+    const canvas = this.node;
+
+    this.node.addEventListener('mousedown', e => this.mouseDown(e));
+    this.node.addEventListener('mousemove', e => this.mouseXY(e));
+    document.body.addEventListener('mouseup', e => this.mouseUp(e));
```

Comment:
Is there a reason these are bound using arrow functions?

I think these should be bound as:
```
document.body.addEventListener('mouseup', this.mouseUp);
```

Anonymous functions cannot be unbound